react/utils/metrics.ts

```diff
@@ -0,0 +1,95 @@
+import axios from 'axios'
+
+const ANALYTICS_URL = 'https://rc.vtex.com/api/analytics/schemaless-events'
+
+type UserData = {
+  id: string
+  canImpersonate: boolean
+  orgId: string
+  clId: string
+  costId: string
+  roleId: string
+}
+
+type MetricsParam = {
+  sessionProfile: SessionProfile
+  accountName: string
+  userData: UserData
+  costCenterName: string
+  buyOrgName: string
+  quoteId: string
+  quoteReferenceName: string
+}
+
+type Metric = {
+  name: 'b2b-suite-buyerorg-data'
+  kind: 'create-quote-ui-event'
+  description: 'Create Quotation Action - UI'
+  account: string
+}
+
+type QuoteFieldsMetric = {
+  org_id: string
+  cost_center_id: string
+  cost_center_name: string
+  buy_org_id: string
+  buy_org_name: string
+  member_id: string
+  member_email: string
+  role: string
+  creation_date: string
+  quote_id: string
+  quote_reference_name: string
+  send_to_sales_rep: boolean
+}
+
+export type SessionProfile = {
+  id: { value: string }
+  email: { value: string }
+
+  firstName?: string
+  lastName?: string
+}
+
+type QuoteMetric = Metric & { fields: QuoteFieldsMetric }
+
+const buildQuoteMetric = (
+  sendToSalesRep: boolean,
+  metricsParam: MetricsParam
+): QuoteMetric => {
+  const metric: QuoteMetric = {
+    name: 'b2b-suite-buyerorg-data',
+    kind: 'create-quote-ui-event',
+    description: 'Create Quotation Action - UI',
+    account: metricsParam.accountName,
+    fields: {
+      org_id: metricsParam.userData?.orgId,
+      cost_center_id: metricsParam.userData?.costId,
+      cost_center_name: metricsParam.costCenterName,
+      buy_org_id: metricsParam.userData.orgId,
+      buy_org_name: metricsParam.buyOrgName,
+      member_id: metricsParam.sessionProfile?.id?.value,
+      member_email: metricsParam.sessionProfile?.email.value,
+      role: metricsParam.userData?.roleId,
+      creation_date: new Date().toISOString(),
+      quote_id: metricsParam.quoteId,
+      quote_reference_name: metricsParam.quoteReferenceName,
+      send_to_sales_rep: sendToSalesRep,
+    },
+  }
+
+  return metric
+}
+
+export const sendMetric = async (
+  sendToSalesRep: boolean,
+  metricsParam: MetricsParam
+) => {
+  try {
+    const metric = buildQuoteMetric(sendToSalesRep, metricsParam)
+
+    axios.post(ANALYTICS_URL, metric)
```

Comment:
In this case, I'm not sure I need to catch errors. The idea of not having an await is not having to wait for the response from analytics API and not delay a bit more the response for users.

Reply:
I agree that it's not very important for you to catch errors here, but in that case you should remove the try/catch. Without the await it won't work well.

If you want to keep catching the errors here you can add the await. That won't really change the behavior of the call, because you don't await the `sendMetric` call itself, so it will still run in the background without affecting the rendering thread in any way.